app_dart/lib/src/service/scheduler.dart

```diff
@@ -450,7 +454,33 @@ class Scheduler {
             success = true;
           }
         } else {
-          await luciBuildService.rescheduleUsingCheckRunEvent(checkRunEvent);
+          final String sha = checkRunEvent.checkRun!.headSha!;
+          final String checkName = checkRunEvent.checkRun!.name!;
+          final RepositorySlug slug = checkRunEvent.repository!.slug();
+
+          // TODO: Can't access branch from [checkRunEvent.checkRun.checkSuite] because head_branch is not deserialized
+          // See https://docs.github.com/developers/webhooks-and-events/webhooks/webhook-events-and-payloads?actionType=rerequested#check_run
+          final String gitBranch = Config.defaultBranch(slug);
```

Comment:
If this is the case, should it be made clear that these only ship for the default branch?

Reply:
So, I'm not clear on whether we can expect postsubmit checkruns to ever happen on a non-primary/default branch. But yea, it would be ideal if this hack was not needed. The main issue is that our CheckRunEvent model embeds a CheckSuite model that is vended by the github package, so I'm not sure yet of the best way to modify it to get the `head_branch` field piped from the response.